language/move-lang/tests/functional/evaluation_order/short_circuiting.move

```diff
@@ -0,0 +1,22 @@
+module X {
+    public fun error(): bool {
+        abort 42
+    }
+}
+
+//! new-transaction
+use {{default}}::X;
+fun main() {
+    let vtrue = true;
+    let vfalse = false;
+
+    true || X::error();
+    vtrue || X::error();
+    vtrue || { let r = X::error(); r };
+    { let x = vtrue; x} || X::error();
+
+    false && X::error();
+    vfalse && X::error();
+    vfalse && { let r = X::error(); r };
+    { let x = vfalse; x} && X::error();
+}
```

Comment:
Are there already tests to check that we 
- evaluate the RHS when the LHS is false for or/true for and
- evaluate the LHS in all cases (not that this is a likely mistake)?

Reply:
> evaluate the RHS when the LHS is false for or/true for and
> evaluate the LHS in all cases (not that this is a likely mistake)

I'll add them to be safe